custom_components/ltss/__init__.py

```diff
@@ -267,46 +267,77 @@ def event_listener(self, event):
             if self.entity_filter(entity_id):
                 self.queue.put(event)
 
+    def _ltss_is_hyptertable(self):
+        if 'timescaledb' not in self.available_extensions:
+            return False
+
+        return 1 == self.con.execute(
+            text(
+                f"SELECT 1 FROM timescaledb_information.hypertables "
+                f"WHERE hypertable_name = '{LTSS.__tablename__}'")).rowcount
+
+    def _ltss_with_postgis(self):
+        return 1 == self.con.execute(
+            text(
+                f"SELECT 1 FROM information_schema.columns "
+                f"WHERE table_name = '{LTSS.__tablename__}' AND COLUMN_NAME = 'location'")).rowcount
+
+    def _update_chunk_time_interval(self):
+        self.con.execute(text(f"SELECT set_chunk_time_interval('{LTSS.__tablename__}', {self.chunk_time_interval});")
+                         .execution_options(autocommit=True)
+                         )
+
     def _setup_connection(self):
         """Ensure database is ready to fly."""
 
         if self.engine is not None:
             self.engine.dispose()
 
         self.engine = create_engine(self.db_url, echo=False,
-            json_serializer=lambda obj: json.dumps(obj, cls=JSONEncoder))
+                                    json_serializer=lambda obj: json.dumps(obj, cls=JSONEncoder))
 
-        # Make sure TimescaleDB  and PostGIS extensions are loaded
-        with self.engine.connect() as con:
-            con.execute(
-                text("CREATE EXTENSION IF NOT EXISTS postgis CASCADE"
-                ).execution_options(autocommit=True))
-            con.execute(
-                text("CREATE EXTENSION IF NOT EXISTS timescaledb CASCADE"
-                ).execution_options(autocommit=True))
+        self.con = self.engine.connect()
 
-        # Create all tables if not exists
-        Base.metadata.create_all(self.engine)
+        self.available_extensions = [row['name'] for row in
+                                     self.con.execute(text("SELECT name FROM pg_available_extensions"))]
+
+        inspector = inspect(self.engine)
+        if not inspector.has_table(LTSS.__tablename__):
+            self._create_table()
+
+        if self._ltss_is_hyptertable():
+            self._update_chunk_time_interval()
+
+        if self._ltss_with_postgis():
+            LTSS.activate_location_extraction()
```

Comment:
This is done twice? Here and on line 326?

Reply:
Yes, that happens twice:
1) Within the `_setup_connection()` it needs to be activated so new entries are created correctly (with the location extraction).
2) Within the `_create_table()` it is necessary so the table is created with the additional column.

So, only on the first start (when the table is created), `activate_location_extraction()` is called twice - and that's no problem.